WorkloadManagementSystem/Agent/JobAgent.py

```diff
@@ -177,28 +180,38 @@ def execute(self):
     result = self.computingElement.getDescription()
     if not result['OK']:
       return result
-    ceDict = result['Value']
-    # Add pilot information
-    gridCE = gConfig.getValue('LocalSite/GridCE', 'Unknown')
-    if gridCE != 'Unknown':
-      ceDict['GridCE'] = gridCE
-    if 'PilotReference' not in ceDict:
-      ceDict['PilotReference'] = str(self.pilotReference)
-    ceDict['PilotBenchmark'] = self.cpuFactor
-    ceDict['PilotInfoReportedFlag'] = self.pilotInfoReportedFlag
-
-    # Add possible job requirements
-    result = gConfig.getOptionsDict('/AgentJobRequirements')
-    if result['OK']:
-      requirementsDict = result['Value']
-      ceDict.update(requirementsDict)
-      self.log.info('Requirements:', requirementsDict)
-
-    self.log.verbose(ceDict)
-    start = time.time()
-    jobRequest = MatcherClient().requestJob(ceDict)
-    matchTime = time.time() - start
-    self.log.info('MatcherTime = %.2f (s)' % (matchTime))
+
+    # We can have several prioritized job retrieval strategies
+    if isinstance(result['Value'], dict):
+      ceDictList = [result['Value']]
+    elif isinstance(result['Value'], list):
+      ceDictList = result['Value']
+
+    for ceDict in ceDictList:
+
+      # Add pilot information
+      gridCE = gConfig.getValue('LocalSite/GridCE', 'Unknown')
+      if gridCE != 'Unknown':
+        ceDict['GridCE'] = gridCE
+      if 'PilotReference' not in ceDict:
+        ceDict['PilotReference'] = str(self.pilotReference)
+      ceDict['PilotBenchmark'] = self.cpuFactor
+      ceDict['PilotInfoReportedFlag'] = self.pilotInfoReportedFlag
+
+      # Add possible job requirements
+      result = gConfig.getOptionsDict('/AgentJobRequirements')
+      if result['OK']:
+        requirementsDict = result['Value']
+        ceDict.update(requirementsDict)
+        self.log.info('Requirements:', requirementsDict)
+
+      self.log.verbose(ceDict)
+      start = time.time()
+      jobRequest = MatcherClient().requestJob(ceDict)
+      matchTime = time.time() - start
+      self.log.info('MatcherTime = %.2f (s)' % (matchTime))
+      if jobRequest['OK']:
+        break
```

Comment:
Why?

Reply:
If you got a job, you exit the loop on different job request strategies and continue with the usual JobAgent logic to process the received job.